src/ApiCore/TransportFactory.php

```diff
@@ -226,33 +221,35 @@ private static function handleTransport($transport)
 
     /**
      * @param array $args
-     * @return CredentialsLoader
+     * @return AuthWrapper
      */
-    private static function handleCredentialsLoader(array $args)
+    private static function buildAuthWrapper(array $args)
     {
+        $authHttpHandler = $args['authHttpHandler'] ?: HttpHandlerFactory::build();
+
         if (isset($args['credentialsLoader'])) {
-            return $args['credentialsLoader'];
-        }
+            $fetchAuthTokenInterface = $args['credentialsLoader'];
```

Comment:
Since this isn't an interface, but an implementation, what do you think about calling it something like `fetcher` or `credentialsLoader`?